pkg/qemu/qemu.go

```diff
@@ -243,6 +247,64 @@ func showDarwinARM64HVFQEMU620Warning(exe, accel string, features *features) {
 	logrus.Warn(w)
 }
 
+func getMacOSProductVersion() (*semver.Version, error) {
+	cmd := exec.Command("sw_vers", "-productVersion")
+	// output is like "12.3.1\n"
+	b, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute %v: %w", cmd.Args, err)
+	}
+	verTrimmed := strings.TrimSpace(string(b))
+	verSem, err := semver.NewVersion(verTrimmed)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse macOS version %q: %w", verTrimmed, err)
+	}
+	return verSem, nil
+}
+
+// adjustMemBytesDarwinARM64HVF adjusts the memory to be less than 4 GiB, only when the following conditions are met:
+//
+// - memBytes  >= 4 GiB
+// - Host OS   <  macOS 12.4
+// - Host Arch == arm64
+// - Accel     == hvf
+// - QEMU      >= 7.0
+//
+// This adjustment is required for avoiding host kernel panic. The issue was fixed in macOS 12.4 Beta 1.
+// See https://github.com/lima-vm/lima/issues/795 https://gitlab.com/qemu-project/qemu/-/issues/903#note_911000975
+func adjustMemBytesDarwinARM64HVF(memBytes int64, accel string, features *features) int64 {
+	if memBytes < (1 << 32) { // less than 4GiB
```

Comment:
Doesn't this one have to be adjusted to 3GiB as well?

Reply:
adjusted to 3 GiB